src/algebra/ordered_monoid.lean

```diff
@@ -42,13 +41,28 @@ class ordered_comm_monoid (α : Type*) extends comm_monoid α, partial_order α
   * `a ≤ b → c + a ≤ c + b` (addition is monotone)
   * `a + b < a + c → b < c`.
 -/
+
 @[protect_proj, ancestor add_comm_monoid partial_order]
 class ordered_add_comm_monoid (α : Type*) extends add_comm_monoid α, partial_order α :=
 (add_le_add_left       : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
 (lt_of_add_lt_add_left : ∀ a b c : α, a + b < a + c → b < c)
 
 attribute [to_additive] ordered_comm_monoid
 
+/-- An `ordered_comm_monoid` with one-sided 'division' in the sense that
+if `a ≤ b`, there is some `c` for which `a * c = b`. This is a weaker
+version of the canonical orderings defined later.   -/
+class has_exists_mul_of_le (α : Type u) [ordered_comm_monoid α] :=
+(exists_mul_of_le : ∀ (a b : α), a ≤ b → ∃ (c : α), b = a * c)
+
+/-- An `ordered_add_comm_monoid` with one-sided 'subtraction' in the sense that
+if `a ≤ b`, then there is some `c` for which `a * c = b`. This is a weaker
+version of the canonical orderings defined later. -/
+class has_exists_add_of_le (α : Type u) [ordered_add_comm_monoid α] :=
+(exists_add_of_le : ∀ (a b : α), a ≤ b → ∃ (c : α), b = a + c)
```

Comment:
I think these might be better as
```suggestion
class has_exists_mul_of_le (α : Type u) [ordered_comm_monoid α] :=
(exists_mul_of_le : ∀ {a b : α}, a ≤ b → ∃ (c : α), b = a * c)

/-- An `ordered_add_comm_monoid` with one-sided 'subtraction' in the sense that
if `a ≤ b`, then there is some `c` for which `a * c = b`. This is a weaker
version of the canonical orderings defined later. -/
class has_exists_add_of_le (α : Type u) [ordered_add_comm_monoid α] :=
(exists_add_of_le : ∀ {a b : α}, a ≤ b → ∃ (c : α), b = a + c)
```
which will remove the need for underscores in the caller.